params.go

```diff
@@ -30,6 +30,58 @@ type Params struct {
 	Data interface{} `json:"-"`
 }
 
+// CommonParams should be used where there are no operation specific Params defined
+type CommonParams struct {
+	Params
+}
+
+func (list *CommonParams) toParams() *Params {
+	return &Params{
+		IdempotencyKey: list.IdempotencyKey,
+		Header:         list.Header,
+		Context:        list.Context,
+		RequestParams:  list,
+	}
+}
+
+// Option allows for customizations to a request.
+type Option func(p *Params)
+
+// CommonParamsWithOptions will create a CommonParams and apply the passed Options.
+func CommonParamsWithOptions(opts []Option) *CommonParams {
```

Comment:
I like this pattern of just creating the params with the options all at once. let's provide a similar function for just Params, i noted this on the test file for more details